src/passes/OptimizeInstructions.cpp

```diff
@@ -2390,6 +2398,10 @@ struct OptimizeInstructions
   }
 
   bool isSymmetricOrRelational(Binary* binary) {
+    if ((binary->op == SubInt32 || binary->op == SubInt64) &&
+        binary->right->is<Const>() && !binary->left->is<Const>()) {
+      return true;
```

Comment:
this change looks confusing: `Sub` is neither symmetric nor relational. Perhaps we should rename the function, if it makes sense to consider a `Sub` as well in all the callers? (Why does that make sense?)

Reply:
It's required for pass this guard for `x - C` case which will be canonicalized to `x + (-C)`. So perhaps rename this function to something like `bool shouldCanonicalize(Binary* binary)` wdyt?

Reply:
Yes, that name sounds better!